framework/src/application/genesis_block.ts

```diff
@@ -65,52 +63,44 @@ export interface GenesisAccountStateJSON {
 	};
 }
 
-const accountFromJSON = (account: GenesisAccountStateJSON): GenesisAccountState<AccountAsset> => ({
-	address: Buffer.from(account.address, 'base64'),
-	balance: BigInt(account.balance),
-	nonce: BigInt(account.nonce),
-	keys: {
-		mandatoryKeys: account.keys.mandatoryKeys.map(key => Buffer.from(key, 'base64')),
-		optionalKeys: account.keys.optionalKeys.map(key => Buffer.from(key, 'base64')),
-		numberOfSignatures: account.keys.numberOfSignatures,
-	},
-	asset: {
-		delegate: {
-			...account.asset.delegate,
-			totalVotesReceived: BigInt(account.asset.delegate.totalVotesReceived),
+export const genesisSchema = (accountSchema: object): Schema => ({
+	...blockSchema,
+	properties: {
+		...blockSchema.properties,
+		header: {
+			...blockHeaderSchema,
+			properties: {
+				...blockHeaderSchema.properties,
+				id: {
+					dataType: 'bytes',
+				},
+				asset: {
+					...genesisBlockHeaderAssetSchema,
+					properties: {
+						...genesisBlockHeaderAssetSchema.properties,
+						accounts: {
+							...(genesisBlockHeaderAssetSchema.properties as { accounts: Record<string, unknown> })
+								.accounts,
+							items: {
+								...baseAccountSchema,
+								properties: {
+									...baseAccountSchema.properties,
+									asset: {
+										...baseAccountSchema.properties.asset,
+										properties: accountSchema,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
```

Comment:
I think this will look easier to read and understand

```js
export const genesisBlockSchema = (accountSchema: object): Schema => mergeDeep({}, 
    blockSchema, 
    genesisBlockHeaderAssetSchema, {
		properties: {
			accounts: {
				items: {
					properties: {
						asset: accountSchema,
					},
				},
			},
		},
	});
```